pipeline/id_minter/src/test/scala/uk/ac/wellcome/platform/idminter/steps/SourceIdentifierEmbedderTest.scala

```diff
@@ -0,0 +1,388 @@
+        |        "ontologyType": "${sourceIdentifier2.ontologyType}"
+        |      },
+        |      "identifiedType": "AnotherNewType"
+        |    }
+        |  ]
+        |}
+      """.stripMargin
+      val json = parse(jsonString).right.get
+      val identifiers = Map(
+        sourceIdentifier1 -> Identifier(createCanonicalId, sourceIdentifier1),
+        sourceIdentifier2 -> Identifier(createCanonicalId, sourceIdentifier2)
+      )
+      val identified = SourceIdentifierEmbedder.update(
+        json,
+        identifiers
+      )
+
+      identified shouldBe a[Success[_]]
+      val updatedJsonString = identified.get.spaces2
+      assertJsonStringsAreEqual(
+        updatedJsonString,
+        s"""
+           |{
+           |  "sourceIdentifier": {
+           |    "identifierType": {
+           |      "id": "${sourceIdentifier1.identifierType.id}",
+           |      "label": "${sourceIdentifier1.identifierType.label}",
+           |      "ontologyType": "IdentifierType"
+           |    },
+           |    "value": "${sourceIdentifier1.value}",
+           |    "ontologyType": "${sourceIdentifier1.ontologyType}"
+           |  },
+           |  "canonicalId": "${identifiers(sourceIdentifier1).CanonicalId}",
+           |  "type": "NewType",
+           |  "moreThings": [
+           |    {
+           |      "sourceIdentifier": {
+           |        "identifierType": {
+           |          "id": "${sourceIdentifier2.identifierType.id}",
+           |          "label": "${sourceIdentifier2.identifierType.label}",
+           |          "ontologyType": "IdentifierType"
+           |        },
+           |        "value": "${sourceIdentifier2.value}",
+           |        "ontologyType": "${sourceIdentifier2.ontologyType}"
+           |      },
+           |      "canonicalId": "${identifiers(sourceIdentifier2).CanonicalId}",
+           |      "type": "AnotherNewType"
+           |    }
+           |  ]
+           |}
+      """.stripMargin
+      )
+    }
+
+    it(
+      "fails if it cannot match the identifier to any sourceIdentifier in the json") {
+      val sourceIdentifier = createSourceIdentifier
+      val otherSourceIdentifier = createSourceIdentifier
+      val jsonString = s"""
+        |{
+        |  "sourceIdentifier": {
+        |    "identifierType": {
+        |      "id": "${sourceIdentifier.identifierType.id}",
+        |      "label": "${sourceIdentifier.identifierType.label}",
+        |      "ontologyType": "IdentifierType"
+        |    },
+        |    "value": "${sourceIdentifier.value}",
+        |    "ontologyType": "${sourceIdentifier.ontologyType}"
+        |  }
+        |}
+      """.stripMargin
+      val json = parse(jsonString).right.get
+      val identifier =
+        createSQLIdentifierWith(sourceIdentifier = otherSourceIdentifier)
+      val identified = SourceIdentifierEmbedder.update(
+        json,
+        Map(otherSourceIdentifier -> identifier))
```

Comment:
Do we need the `otherSourceIdentifier` here? AIUI, it’s walking the JSON and trying to look up the `sourceIdentifier` in the map of identifiers it receives – would `Map.empty` work too?